core/domain/learner_group_services.py

```diff
@@ -0,0 +1,647 @@
+            part of the learner group syllabus. Each subtopic page id is
+            represented as a topicId:subtopicId string.
+        story_ids: list(str). The ids of the stories that are part of the
+            learner group syllabus.
+
+    Returns:
+        learner_group: learner_group_domain.LearnerGroup. The domain object
+        of the updated learner group.
+
+    Raises:
+        Exception. The learner group with the given id does not exist.
+    """
+
+    learner_group_model = learner_group_models.LearnerGroupModel.get(
+        group_id, strict=False)
+
+    if not learner_group_model:
+        raise Exception(
+            'The learner group with the given group id does not exist.')
+
+    old_invited_student_ids = set(learner_group_model.invited_student_user_ids)
+    new_invited_student_ids = set(invited_student_ids)
+    if new_invited_student_ids != old_invited_student_ids:
+        newly_added_invites = list(
+            new_invited_student_ids - old_invited_student_ids
+        )
+        newly_removed_invites = list(
+            old_invited_student_ids - new_invited_student_ids
+        )
+        invite_students_to_learner_group(
+            group_id, newly_added_invites)
+        remove_invited_students_from_learner_group(
+            group_id, newly_removed_invites)
+
+    learner_group_model.title = title
+    learner_group_model.description = description
+    learner_group_model.facilitator_user_ids = facilitator_user_ids
+    learner_group_model.student_user_ids = student_ids
+    learner_group_model.invited_student_user_ids = invited_student_ids
+    learner_group_model.subtopic_page_ids = subtopic_page_ids
+    learner_group_model.story_ids = story_ids
+
+    learner_group_model.update_timestamps()
+    learner_group_model.put()
+
+    learner_group = get_learner_group_from_model(learner_group_model)
+    learner_group.validate()
+
+    return learner_group
+
+
+def is_user_facilitator(user_id: str, group_id: str) -> bool:
+    """Checks if the user is a facilitator of the leaner group.
+
+    Args:
+        user_id: str. The id of the user.
+        group_id: str. The id of the learner group.
+
+    Returns:
+        bool. Whether the user is a facilitator of the learner group.
+    """
+    learner_group = learner_group_fetchers.get_learner_group_by_id(
+        group_id)
+
+    # Ruling out the possibility of None for mypy type checking.
+    assert learner_group is not None
+
+    return user_id in learner_group.facilitator_user_ids
+
+
+def remove_learner_group(group_id: str) -> None:
+    """Removes the learner group with of given learner group ID.
+
+    Args:
+        group_id: str. The id of the learner group to be removed.
+    """
+    learner_group_model = learner_group_models.LearnerGroupModel.get(
+        group_id, strict=False)
+
+    # Ruling out the possibility of None for mypy type checking.
+    assert learner_group_model is not None
```

Comment:
If you expect the `learner_group_model` to always exist call it with `strict=True`

Reply:
Done!